src/main/java/ua/dymohlo/hibernetpractic/Main.java

```diff
@@ -0,0 +1,162 @@
+package ua.dymohlo.hibernetpractic;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.query.Query;
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+
+
+public class Main {
+    public static void main(String[] args) {
+        newGraduate();
+        buyersAndOrders();
+        findingCharactersAndCalcOrders();
+    }
+
+    private static void newGraduate() {
```

Comment:
In your CRUD method, you creating the same configuration and creating session factory.

SessionFactory is a general object that should be configured and created only once across all application life-cycle.

You need to configure it somewhere and just get connections(entitymanager/session) from it.

Also, it's bad because you do a code duplication and we can avoid this.

For example:
```
private static SessionFactory sessionFactory;

    static {
        Configuration configuration = new Configuration();
        configuration.configure();
        sessionFactory = configuration.buildSessionFactory();
    }

    private static Session getSession() {
        return sessionFactory.openSession();
    }
    ```
    
    ```
    private static void newGraduate() {
        try (Session session = getSession()) {
            beginTransaction(session);

            // ... rest of the method code ...
            
           }
````